### PR TITLE
Add coverage reporting.

### PR DESCRIPTION
**Pull request by asakatida**
- [x] I have read and agreed to the RethinkDB Contributor License Agreement http://rethinkdb.com/community/cla/

### Description

Adds Travis steps to export coverage data to Coveralls.

### CODE REVIEW

**Review by ralyodio — APPROVED** [review]
lgtm

**Comment by asakatida** [comment]
I still need to come back to this and explore why coveralls is pending.

**Comment by asakatida** [comment]
Rebased to remove v8 diffs from this PR.

File: README.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 [RethinkDB](https://www.rethinkdb.com)
 ======================================
 
+[![Coverage Status](https://coveralls.io/repos/github/RebirthDB/rebirthdb/badge.svg?branch=next)](https://coveralls.io/github/RebirthDB/rebirthdb?branch=next)
```
Comment (floydkots) [inline README.md]: @grandquista, if it won't be too far outside this PR's scope, would you add the Travis build status badge too?

**Comment by atris** [comment]
Looks good, I can merge now unless @grandquista wants to address @floydkots comments

**Comment by asakatida** [comment]
Addressed the missing badge. I'm good with this being merged now.

**Comment by gabor-boros** [comment]
Can we merge this PR?

**Review by atris — APPROVED** [review]

**Pull request merged**